src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+
+        Returns:
+            tuple: The model and optimizer and scheduler.
+        """
+        model = self._get(self.config_init, "model")
+        optimizer_kwargs = self._get(self.config_init, "optimizer", {})
+        if isinstance(optimizer_kwargs, Namespace):
+            optimizer_kwargs = namespace_to_dict(optimizer_kwargs)
+        scheduler_kwargs = self._get(self.config_init, "scheduler", {})
+        if isinstance(scheduler_kwargs, Namespace):
+            scheduler_kwargs = namespace_to_dict(scheduler_kwargs)
+        from otx.core.utils.instantiators import partial_instantiate_class
+
+        return model, partial_instantiate_class(optimizer_kwargs), partial_instantiate_class(scheduler_kwargs)
+
+    def _get(self, config: Namespace, key: str, default: Any = None) -> Any:  # noqa: ANN401
+        """Utility to get a config value which might be inside a subcommand."""
+        return config.get(str(self.subcommand), config).get(key, default)
+
+    def _parser(self, subcommand: str | None) -> ArgumentParser:
+        if subcommand is None:
+            return self.parser
+        # return the subcommand parser for the subcommand passed
+        return self._subcommand_parsers[subcommand]
+
+    def _prepare_subcommand_kwargs(self, subcommand: str) -> dict[str, Any]:
+        """Prepares the keyword arguments to pass to the subcommand to run."""
+        return {
+            k: v for k, v in self.config_init[subcommand].items() if k in self._subcommand_method_arguments[subcommand]
+        }
+
+    def save_config(self) -> None:
+        """Save the configuration for the specified subcommand.
+
+        The configuration is saved as a YAML file in the engine's working directory.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        self._parser(self.subcommand).save(
+            cfg=self.config.get(str(self.subcommand), self.config),
+            path=Path(self.engine.work_dir) / "configs.yaml",
+            overwrite=True,
+            multifile=False,
+            skip_check=True,
+        )
+
+    def run(self) -> None:
+        """Executes the specified subcommand.
+
+        Raises:
+            ValueError: If the subcommand is not recognized.
+        """
+        self.console.print(f"[blue]{OTX_LOGO}[/blue] ver.{__version__}", justify="center")
+        if self.subcommand == "install":
+            from otx.cli.install import otx_install
+
+            otx_install(**self.config["install"])
+        elif self.subcommand in self.engine_subcommands():
+            self.instantiate_classes()
+            fn_kwargs = self._prepare_subcommand_kwargs(self.subcommand)
+            fn = getattr(self.engine, self.subcommand)
+            try:
+                fn(**fn_kwargs)
+            except Exception:
+                self.console.print_exception(width=self.console.width)
+            self.save_config()
+        else:
+            msg = f"Unrecognized subcommand: {self.subcommand}"
```

Comment:
is it possible to list the available subcommands as well?

Reply:
I think the install and engine subcommands should appear fluid, I'll try to think of a way to do this.